src/main/java/ch/njol/skript/registrations/EventValues.java

```diff
@@ -454,12 +457,31 @@ public static boolean doesEventValueHaveTimeStates(Class<? extends Event> event,
 		return getEventValueConverter(event, c, TIME_PAST, false) != null || getEventValueConverter(event, c, TIME_FUTURE, false) != null;
 	}
 
-	private record EventValueInfo<E extends Event, T>(
+	/**
+	 * @return All the event values for each registered event's class.
+	 */
+	public static Multimap<Class<? extends Event>, EventValueInfo<?, ?>> getPerEventEventValues() {
+		Multimap<Class<? extends Event>, EventValueInfo<?, ?>> eventValues = MultimapBuilder
+			.hashKeys()
+			.hashSetValues()
+			.build();
+
+		for (int i = TIME_PAST; i <= TIME_FUTURE; i++) {
+			for (EventValueInfo<?, ?> eventValueInfo : getEventValuesListForTime(i)) {
+				Collection<EventValueInfo<?, ?>> existing = eventValues.get(eventValueInfo.event);
+				existing.add(eventValueInfo);
+				eventValues.putAll(eventValueInfo.event, existing);
+			}
+		}
+		return eventValues;
+	}
+
+	public record EventValueInfo<E extends Event, T>(
 		Class<E> event, Class<T> c, Converter<E, T> converter,
 		@Nullable String excludeErrorMessage,
 		@Nullable Class<? extends E>[] excludes
 	) {
-		private EventValueInfo {
+		public EventValueInfo {
```

Comment:
Can this be package access if you're only needing this within the same class?

Reply:
no this is used as the return value for getEventValues